keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane/README.md

```diff
@@ -0,0 +1,960 @@
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+This KEP proposes that the control-plane in kubeadm be run as non-root. If 
+containers are running as root an escape from a container may result in the 
+escalation to root in host. [CVE-2019-5736](https://kubernetes.io/blog/2019/02/11/runc-and-cve-2019-5736/)
+is an example of a container escape vulnerabilty that can be mitigated by 
+running containers/pods as non-root. 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+[CVE-2019-5736](https://kubernetes.io/blog/2019/02/11/runc-and-cve-2019-5736/)
+is an example of a container escape vulnerabilty that can be mitigated by 
+running containers/pods as non-root.
+
+Running containers as non-root has been a long recommended best-practice in 
+kubernetes and we have published [blogs](https://kubernetes.io/blog/2018/07/18/11-ways-not-to-get-hacked/#8-run-containers-as-a-non-root-user) recommending this best practice. kubeadm
+which is a tool built to provide a "best-practice" path to creating clusters is
+an ideal candidate to apply this to.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Run control-plane components as non-root in kubeadm. More specifically :-
+  - Run `kube-apiserver` as non-root in kubeadm.
+  - Run `kube-controller-manager` as non-root in kubeadm.
+  - Run `kube-scheduler` as non-root in kubeadm.
```

Comment:
i'd say, we have to include the etcd static pod here too.

Reply:
Done.